.github/actions/setup_python/action.yml

```diff
@@ -0,0 +1,67 @@
+name: 'Setup Python and pip cache'
+description: 'Setups Python with the provided version and sets up the pip cache'
+inputs:
+  version:
+    description: 'Python version to install'
+    required: true
+  pip-cache-path:
+    description: 'Path on share where the pip cache is stored'
+    required: false
+  should-setup-pip-paths:
+    description: 'If the action should setup `PIP_CACHE_DIR` & `PIP_INSTALL_PATH` env variables'
+    required: false
+    default: 'false'
+  self-hosted-runner:
+    description: 'If the runner is self-hosted'
+    required: false
+    default: 'true'
+runs:
+  using: 'composite'
+  steps:
+
+    - if: ${{ runner.os == 'Linux' && inputs.self-hosted-runner == 'true' }}
+      name: Install 'actions/setup-python@v4' dependencies
+      shell: bash
+      run: |
+        # Fedora
+        if [[ -e /etc/fedora-release ]]; then
+          echo "Inside a Fedora container, no need to install dependencies."
+        else
+          apt-get update && apt-get install -y ca-certificates
+        fi
+
+    - if: ${{ runner.os == 'Linux' && runner.arch == 'ARM64' }}
+      name: Setup sudo
+      shell: bash
+      run: apt-get update && apt-get install -y sudo # Needed for the deadsnakes action
+
+    - if: ${{ runner.os == 'Linux' && runner.arch == 'ARM64' }}
+      name: Setup Python ${{ inputs.version }}
+      uses: deadsnakes/action@v3.0.1
+      with:
+        python-version: ${{ inputs.version }}
+
+    - if: ${{ runner.os == 'macOS' || runner.os == 'Windows' || (runner.os == 'Linux' && runner.arch != 'ARM64') }}
+      name: Setup Python ${{ inputs.version }}
```

Comment:
To solve the following warning:
WARNING: The directory '/github/home/.cache/pip' or its parent directory is not owned or is not writable by the current user. The cache has been disabled. Check the permissions and owner of that directory. If executing pip with sudo, you should use sudo's -H flag.

please set the available cache directory first (e.g. ${{ inputs.pip-cache-path }})